pkg/groupsnapshot/controllers/groupsnapshot.go

```diff
@@ -328,16 +328,34 @@ func (m *GroupSnapshotController) handleSnap(groupSnap *stork_api.GroupVolumeSna
 	}
 
 	if isFailed, failedTasks := isAnySnapshotFailed(response.Snapshots); isFailed {
-		err = fmt.Errorf("Some snapshots in group have failed: %s."+
-			" Resetting group snapshot to retry.", failedTasks)
+		errMsgPrefix := fmt.Sprintf("Some snapshots in group have failed: %s", failedTasks)
+
+		if groupSnap.Status.NumRetries < groupSnap.Spec.MaxRetries {
+			groupSnap.Status.NumRetries++
+
+			err = fmt.Errorf("%s. Resetting group snapshot for retry: %d",
+				errMsgPrefix, groupSnap.Status.NumRetries)
+			response.Snapshots = nil // so that snapshots are retried
```

Comment:
This is not enough. If the storage driver is still backing up some of the snapshots then those need to be canceled and deleted.

Reply:
That's already being done in the driver: https://github.com/libopenstorage/stork/tree/master/https://github.com/libopenstorage/stork/blob/master/drivers/volume/portworx/portworx.go#L1513

This PR doesn't change that behavior and just disables the default retries. `response.Snapshots` was getting set to nil before this PR when the controller was always retrying.

Reply:
Ok, then that should be moved to the controller here from the driver. Status() APIs should be read-only, they shouldn't modify any state. We should add an API to the volume driver interface to cancel group snapthots that should be called from the controller in that case. 

Doesn't need to be done with this change. Can you file a bug for 2.1.1 to fix this?

Reply:
#268 created for that.